website/www/site/content/en/documentation/dsls/dataframes/overview.md

```diff
@@ -0,0 +1,131 @@
+  | beam.Map(lambda row: f"{row.passenger_count}: {row.trip_distance}")
+  ...
+{{< /highlight >}}
+
+`DataframeTransform` is similar to `SqlTransform` from the [Beam SQL](https://beam.apache.org/documentation/dsls/sql/overview/) DSL. Where `SqlTransform` translates a SQL query to a PTransform, `DataframeTransform` is a PTransform that applies a function that takes and returns DataFrames. A `DataframeTransform` can be particularly useful if you have a stand-alone function that can be called both on Beam and on ordinary Pandas DataFrames.
+
+`DataframeTransform` can accept and return multiple PCollections by name and by keyword, as shown in the following examples:
+
+{{< highlight py >}}
+output = (pc1, pc2) | DataframeTransform(lambda df1, df2: ...)
+
+output = {'a': pc, ...} | DataframeTransform(lambda a, ...: ...)
+
+pc1, pc2 = {'a': pc} | DataframeTransform(lambda a: expr1, expr2)
+
+{...} = {a: pc} | DataframeTransform(lambda a: {...})
+{{< /highlight >}}
+
+## Differences from standard Pandas {#differences_from_standard_pandas}
+
+The Beam DataFrames API is deferred, like the rest of the Beam API. As a result, there are some limitations on what you can do with Beam DataFrames, compared to the standard Pandas implementation:
+
+* Because all operations are deferred, the result of a given operation may not be available for control flow. For example, you can compute a sum, but you can't branch on the result.
+* Result columns must be computable without access to the data. For example, you can’t use [transpose](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.transpose.html).
+* PCollections in Beam are inherently unordered, so Pandas operations that are sensitive to the ordering of rows are unsupported.
+
+With Beam DataFrames, computation doesn’t take place until the pipeline runs. Before that, only the shape or schema of the result is known, meaning that you can work with the names and types of the columns, but not the result data itself.
+
+There are a few common exceptions you may see when attempting to use certain Pandas operations:
+
+* **WontImplementError**: Indicates that this is an operation or argument that isn’t supported because it’s incompatible with the Beam model. The largest class of operations that raise this error are operations that are order sensitive, such as [shift](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.shift.html), [cummax](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.cummax.html), [cummin](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.cummin.html), [head](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.head.html), and [tail](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.tail.html#pandas.DataFrame.tail). These cannot be trivially mapped to Beam because PCollections, representing distributed datasets, are unordered.
```

Comment:
```suggestion
* **WontImplementError**: Indicates that this is an operation or argument that we do not intend to support because of one of the limitations listed above.
```

I think the detail about order-sensitive operations is good, but it should be moved up to the "unordered" limitation above.